Havtorn/Source/Editor/EditorManager.cpp

```diff
@@ -137,20 +137,20 @@ namespace Havtorn
 	void CEditorManager::SetCurrentScene(CScene* scene)
 	{
 		CurrentScene = scene;
-		SelectedEntity = nullptr;
+		SelectedEntity = SEntity::Null;
 	}
 
 	CScene* CEditorManager::GetCurrentScene() const
 	{
 		return CurrentScene;
 	}
 
-	void CEditorManager::SetSelectedEntity(SEntity* entity)
+	void CEditorManager::SetSelectedEntity(const SEntity& entity)
 	{
 		SelectedEntity = entity;
 	}
 
-	SEntity* CEditorManager::GetSelectedEntity() const
+	SEntity CEditorManager::GetSelectedEntity() const
```

Comment:
Should we set the return type to also be const? And maybe even a reference as well?
Since the setter accepts a const SEntity&
(Although I do agree with returning a copy, could be a useful reminder that SEntity is basically just an identifier, and holding a specific reference isn't what makes it useful.)
```
void SetSelectedEntity(const SEntity& entity);
```
```suggestion
	const SEntity CEditorManager::GetSelectedEntity() const
```
```suggestion
	const SEntity& CEditorManager::GetSelectedEntity() const
```